Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -43,102 +53,136 @@ FieldProbe::FieldProbe (std::string rd_name)
     pp_amr.query("max_level", nLevel);
     nLevel += 1;
 
+    /* Obtain input data from parsing inputs file.
+     * For the case of a single particle:
+     *     Define x, y, and z of particle
+     *     Define whether or not to integrate fields
+     * For the case of a line detector:
+     *     Define x, y, and z of end of line point 1
+     *     Define x, y, and z of end of line point 2
+     *     Define resoliton to determine number of particles
+     *     Define whether ot not to integrate fields
+     * For the case of a plane detector:
+     *     Define a vector normal to the detector plane
+     *     Define a vector in the "up" direction of the plane
+     *     Define the size of the plane (width of half square)
+     *     Define resoliton to determine number of particles
+     *     Define whether ot not to integrate fields
+     */
     amrex::ParmParse pp_rd_name(rd_name);
     getWithParser(pp_rd_name, "x_probe", x_probe);
-#if (AMREX_SPACEDIM == 3)
     getWithParser(pp_rd_name, "y_probe", y_probe);
-#endif
     getWithParser(pp_rd_name, "z_probe", z_probe);
-
+    getWithParser(pp_rd_name, "integrate", field_probe_integrate);
     pp_rd_name.query("raw_fields", raw_fields);
-
     pp_rd_name.query("interp_order", interp_order);
 
     AMREX_ALWAYS_ASSERT_WITH_MESSAGE(interp_order <= WarpX::nox ,
-                                     "Field probe interp_order should be lower than algo.particle_shape");
+                                     "Field probe interp_order should be lower or equal than algo.particle_shape");
     // resize data array
+
     m_data.resize(noutputs*nLevel, 0.0_rt);
 
     if (ParallelDescriptor::IOProcessor())
     {
         if ( m_IsNotRestart )
         {
+
             // open file
             std::ofstream ofs{m_path + m_rd_name + "." + m_extension, std::ofstream::out};
+
             // write header row
             int c = 0;
             ofs << "#";
             ofs << "[" << c++ << "]step()";
             ofs << m_sep;
             ofs << "[" << c++ << "]time(s)";
-            for (int lev = 0; lev < nLevel; ++lev)
+            if (field_probe_integrate == 0)
```

Comment:
```suggestion
            if (!m_field_probe_integrate)
```
or even better: use `if(m_field_probe_integrate)` and switch the `if` and `else` arms.